atlp-rwanda/knights-ecomm-fe
# Implement Single Product View and Cart management (Guest and User)

#### What does this PR do?
This pull request implement single product view page and cart management 

#### Description of Task to be completed?
- Implemented Skeleton Loader for Both Single Product and Cart
- Viewing single product details
- Changing quantity of product user/guest wish to buy
- Adding Product to Cart
- Updating Product in Cart
- Removing product in cart

#### How should this be manually tested?
- Ensure You have backend service running and set VITE_APP_API_URL in your .env file.
- Clone this repository and navigate to the project directory
    ```
    cd knights-ecomm-fe    
    git fetch
    git checkout ft-cart-managment
    npm i
    npm run dev
    ```
try to view single product and add it to cart

SingleProduct UI Page
![image](https://github.com/atlp-rwanda/knights-ecomm-fe/assets/78812884/3470e3da-99ed-4029-b35c-b282f3b19f75)

Cart UI Page
![image](https://github.com/atlp-rwanda/knights-ecomm-fe/assets/78812884/a5890741-d1d1-49d5-af2c-1fb2226ead66)
